subprojects/boost_math.wrap

```diff
@@ -0,0 +1,7 @@
+[wrap-git]
```

Comment:
I had expected the git submodule to simply move under `subprojects/`, just like for HiGHS.

Reply:
I haven't figured out how to get the

```
[provide]
boost = boost_math_dep
```

working without the wrap file.

Maybe, instead of getting the dep straight from boost_math's CMakeLists, we want to construct our own fallback dependency from the submodule out of the headers we need?

Reply:
Yes, that sounds right, you should be able to do that with `declare_dependency`.

Reply:
Ah, I remember now why we didn't do this:

```
The project is trying to access the directory 'subprojects/boost_math/include' which belongs to a different
subproject. This is a problem as it hardcodes the relative paths of these two projects.
This makes it impossible to compile the project in any other directory layout and also
prevents the subproject from changing its own directory layout.

Instead of poking directly at the internals the subproject should be executed and
it should set a variable that the caller can then use. Something like:

# In subproject
some_dep = declare_dependency(include_directories: include_directories('include'))

# In subproject wrap file
[provide]
some = some_dep

# In parent project
some_dep = dependency('some')
executable(..., dependencies: [some_dep])

This warning will become a hard error in a future Meson release.
```

So meson seems to be pushing us towards the wrap file.